### ISSUE TITLE
PC98 keyboard mapping issues

### ISSUE THREAD

**Issue by sdsnatcher** [joncampbell123/dosbox-x#1310]
On Japanese keyboards, the _ key (underscore) sits between the /? and RSHIFT keys. But this key is not being shown in the keyboard mapper. Instead there are three \ keys. One is named " YEN", but produces the same \ char because on the Japanese charset, the 5Ch character is the Yen symbol.
The 2nd is shown on its correct place on the keyboard, just left of the ENTER key.
And the 3rd redundant \ key is shown on the block of special keys, just under the YEN key.

Here's a [good picture of the PC-98 keyboard](https://commons.wikimedia.org/wiki/File:NEC_PC-9800_series_keyboard.jpg), for reference.

**Comment by joncampbell123** [comment]
I understand the issue, but the mapper code as it is right now is very inflexible with regard to button placement on the user interface as inherited from DOSBox. That is why those keys are outside the keyboard.

**Comment by joncampbell123** [comment]
The mapper layout will be refactored at some point when I have time and a proper design to support multiple layouts.

**Comment by sdsnatcher** [comment]
Ok, I understand that the mapper layout has restrictions. But my point is that the underscore key is just missing, while the \ key is being shown three times. Maybe one of the redundant \ keys could be replaced by the needed _ key, so it can be mapped?

**Comment by joncampbell123** [comment]
There needs to be a _ key, yes. I'll see what I can do.

**Comment by joncampbell123** [comment]
Going over the mapper and the picture of the keyboard, the _ key is the shift state of Ro (ロ), and Ro is there in the mapper.

**Comment by sdsnatcher** [comment]
Indeed, you're right!

I changed the subject of this ticket to a more generic one, to avoid keep opening tickets for each key individually.

> I am open to the idea of changing the text to "Ro / _" instead to clarify that.

That's a good idea, thanks!

I attached here a pc98mapper.txt, where I tried ti assign the US-layout to the PC98 layout the best as possible.

But, as a suggestion dosbox-x should have a way to specify specific mapping files for each machine type. Like:

mapperfile = pc98:pc98mapper.txt, fmtowns:townsmapper.txt, pcjr:pcjrmapper.txt, default:mapper.txt

(The key for the dictionary above would be the contents of the machine entry of dosbox.conf)

[pc98mapper.txt](https://github.com/joncampbell123/dosbox-x/files/3801720/pc98mapper.txt)
